src/jit/codegenarm64.cpp

```diff
@@ -3167,15 +3171,48 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                 varNum = compiler->lvaOutgoingArgSpaceVar;
             }
 
-            if (data->isContained())
+            // Do we have a TYP_STRUCT argument, if so it must be a 16-byte pass-by-value struct
+            if (targetType == TYP_STRUCT)
```

Comment:
That's really target-dependent, and I think again we want something like varTypeIsNonEnregisterableStruct (which currently doesn't exist, but is effectively the same as TYP_STRUCT).